packages/react-sdk/src/components/PresentBar/PresentBar.tsx

```diff
@@ -56,6 +58,23 @@ export function PresentBar() {
   const isPresenting = state.type === 'presenting';
   const isPresentingInEditMode = isPresenting && state.isEditMode;
 
+  const togglePresentationMode = useCallback(() => {
+    const storedGridStatus = localStorage.getItem('showGridState');
+    if (!isPresenting) {
+      if (isShowGrid) {
+        localStorage.setItem('showGridState', 'true');
+        setShowGrid(false);
+      }
+      togglePresentation();
+    } else {
+      if (storedGridStatus === 'true') {
+        setShowGrid(true);
+        localStorage.removeItem('showGridState');
+      }
+      togglePresentation();
```

Comment:
It call `togglePresentation()` in both cases. It could be moved outside of the if statements at the end of the callback. Then also the if conditions can be combined, so that we get less if-blocks.